docs/updates/2018_03_26.md

```diff
@@ -0,0 +1,62 @@
+# Update for March 26, 2018
+
+## Contributions
+
+[@matthewwithanm](https://github.com/matthewwithanm) of Facebook's Nuclide team helped us improve our React game by [avoiding use of deprecated string refs](https://github.com/atom/xray/pull/50) and [avoiding the use of component `state` for data that unrelated to rendering](https://github.com/atom/xray/pull/51). Thanks Matthew!
+
+## The switch to a client/server architecture is complete
+
+We merged [#46](https://github.com/atom/xray/pull/46) last week, completing our switch to a client/server architecture. JavaScript in Xray's user interface now communicates with the Rust core over a domain socket rather than via a native V8 extension, which dramatically simplifies our build process. We connect to the server over a domain socket, which unfortunately means that Xray doesn't work on Windows for now due to the unavailability of domain sockets in the OS. If anyone is interested in adding support for named pipes on Windows to `xray_server`, we'd gladly collaborate on a pull request. If you've tried to build Xray and ran into trouble, now would be a good time to try again on non-Windows platforms after [carefully reading our build instructions](../../CONTRIBUTING.md#building).
+
+## Updated roadmap
+
+We've adjusted our roadmap a bit to prioritize collaborative editing rather than focusing on producing WebAssembly-based editor build. A browser-compatible editor is still part of our long term plan and we're designing the system with that requirement in mind, but since we want all of Xray's features to support remote collaboration, it makes sense to get it into the architecture early.
+
+## Fast file finding
+
+Xray is currently hard-coded to open a single buffer containing the dev build of React, which isn't very useful. To fix that, we're adding a file finder that can quickly filter all files in the project that match a given subsequence query.
+
+To obtain good search performance, we're maintaining an in-memory replica of the names of all the files and directories in the project which we can brute-force scan on a background thread whenever the query changes. We represent this data as a simple tree which reflects the hierarchy of the file system. To ensure that we can respond to user input within our 50ms deadline for coarse-grained interactions, we really want to be able to run queries before we finish reading all of the entries from the file system. To enable that, we're designing our in-memory file tree to support concurrent reads and writes.
+
+We spent a decent amount of time exploring different approaches that could enable this, and ultimately we decided to protect the entry vector for each directory with a fine-grained read/write lock. When [@as-cii](https://github.com/as-cii) first suggested this approach, I was worried that it would consume too much memory, but I then discovered the [parking_lot] crate, whose `RwLock` implementation only consumes a word of memory per instance.
```

Comment:
It seems that the `[parking_lot]` here aims to be with its GitHub repo link?
https://github.com/Amanieu/parking_lot

Reply:
🙇